package org.openapitools.model;

import java.net.URI;
import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonTypeName;
import org.openapitools.model.ConfigNodePropertyBoolean;
import org.openapitools.model.ConfigNodePropertyString;
import org.openapitools.jackson.nullable.JsonNullable;
import java.time.OffsetDateTime;
import jakarta.validation.Valid;
import jakarta.validation.constraints.*;
import io.swagger.v3.oas.annotations.media.Schema;


import java.util.*;
import jakarta.annotation.Generated;

/**
 * ComDayCqWcmCoreImplServletsContentfinderPageViewHandlerProperties
 */

@JsonTypeName("comDayCqWcmCoreImplServletsContentfinderPageViewHandlerProperties")
@Generated(value = "org.openapitools.codegen.languages.JavaCamelServerCodegen", date = "2026-10-07T12:53:52.330827711Z[Etc/UTC]", comments = "Generator version: 7.24.0")
public class ComDayCqWcmCoreImplServletsContentfinderPageViewHandlerProperties {

  private ConfigNodePropertyString guessTotal;

  private ConfigNodePropertyBoolean tagTitleSearch;

  public ComDayCqWcmCoreImplServletsContentfinderPageViewHandlerProperties guessTotal(ConfigNodePropertyString guessTotal) {
    this.guessTotal = guessTotal;
    return this;
  }

  /**
   * Get guessTotal
   * @return guessTotal
   */
  @Valid 
  @Schema(name = "guessTotal", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
  @JsonProperty("guessTotal")
  public ConfigNodePropertyString getGuessTotal() {
    return guessTotal;
  }

  public void setGuessTotal(ConfigNodePropertyString guessTotal) {
    this.guessTotal = guessTotal;
  }

  public ComDayCqWcmCoreImplServletsContentfinderPageViewHandlerProperties tagTitleSearch(ConfigNodePropertyBoolean tagTitleSearch) {
    this.tagTitleSearch = tagTitleSearch;
    return this;
  }

  /**
   * Get tagTitleSearch
   * @return tagTitleSearch
   */
  @Valid 
  @Schema(name = "tagTitleSearch", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
  @JsonProperty("tagTitleSearch")
  public ConfigNodePropertyBoolean getTagTitleSearch() {
    return tagTitleSearch;
  }

  public void setTagTitleSearch(ConfigNodePropertyBoolean tagTitleSearch) {
    this.tagTitleSearch = tagTitleSearch;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    ComDayCqWcmCoreImplServletsContentfinderPageViewHandlerProperties comDayCqWcmCoreImplServletsContentfinderPageViewHandlerProperties = (ComDayCqWcmCoreImplServletsContentfinderPageViewHandlerProperties) o;
    return Objects.equals(this.guessTotal, comDayCqWcmCoreImplServletsContentfinderPageViewHandlerProperties.guessTotal) &&
        Objects.equals(this.tagTitleSearch, comDayCqWcmCoreImplServletsContentfinderPageViewHandlerProperties.tagTitleSearch);
  }

  @Override
  public int hashCode() {
    return Objects.hash(guessTotal, tagTitleSearch);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class ComDayCqWcmCoreImplServletsContentfinderPageViewHandlerProperties {\n");
    sb.append("    guessTotal: ").append(toIndentedString(guessTotal)).append("\n");
    sb.append("    tagTitleSearch: ").append(toIndentedString(tagTitleSearch)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    return o == null ? "null" : o.toString().replace("\n", "\n    ");
  }
}

